benchmark/agbenchmark/__main__.py

```diff
@@ -54,245 +69,335 @@ def get_unique_categories() -> set[str]:
-    if keep_answers:
-        pytest_args.append("--keep-answers")
 
-    if test:
-        print("Running specific test:", test)
+    if tests:
+        logger.info(f"Running specific test(s): {' '.join(tests)}")
+        pytest_args += [f"--test={t}" for t in tests]
     else:
         # Categories that are used in the challenges
-        categories = get_unique_categories()
-        if category:
-            invalid_categories = set(category) - categories
-            assert (
-                not invalid_categories
-            ), f"Invalid categories: {invalid_categories}. Valid categories are: {categories}"
-
-        if category:
-            categories_to_run = set(category)
-            if skip_category:
-                categories_to_run = categories_to_run.difference(set(skip_category))
-                assert categories_to_run, "Error: You can't skip all categories"
-            pytest_args.extend(["-m", " or ".join(categories_to_run), "--category"])
-            print("Running tests of category:", categories_to_run)
-        elif skip_category:
-            categories_to_run = categories - set(skip_category)
+        all_categories = get_unique_categories()
+        if categories:
+            invalid_categories = set(categories) - all_categories
+            assert not invalid_categories, (
+                f"Invalid categories: {invalid_categories}. "
+                f"Valid categories are: {all_categories}"
+            )
+
+        if categories or skip_categories:
+            categories_to_run = set(categories) or all_categories
+            if skip_categories:
+                categories_to_run = categories_to_run.difference(set(skip_categories))
             assert categories_to_run, "Error: You can't skip all categories"
-            pytest_args.extend(["-m", " or ".join(categories_to_run), "--category"])
-            print("Running tests of category:", categories_to_run)
+            pytest_args += [f"--category={c}" for c in categories_to_run]
+            logger.info(f"Running tests of category: {categories_to_run}")
         else:
-            print("Running all categories")
+            logger.info("Running all categories")
 
         if maintain:
-            print("Running only regression tests")
-            pytest_args.append("--maintain")
+            logger.info("Running only regression tests")
         elif improve:
-            print("Running only non-regression tests")
-            pytest_args.append("--improve")
+            logger.info("Running only non-regression tests")
         elif explore:
-            print("Only attempt challenges that have never been beaten")
-            pytest_args.append("--explore")
+            logger.info("Only attempt challenges that have never been beaten")
 
     if mock:
-        pytest_args.append("--mock")
+        # TODO: unhack
         os.environ[
             "IS_MOCK"
         ] = "True"  # ugly hack to make the mock work when calling from API
 
```

Comment:
I would expect this to be handled through a benchmark config. Is that the case

Reply:
Ideally yes, but fixing this is not a priority as I don't think "mock" mode is worth putting a lot of effort into. I'm not even sure why it's in there, but I'll leave it for now until I know more about it or it becomes a problem.